### DISCUSSION TITLE
Help with: Euler's Equations

### DISCUSSION THREAD

**Discussion by nishant-kumar-14** [lululxvi/deepxde#1126] [Q&A]
I'm trying to simulate the following equations:

![Screenshot 2023-01-30 185040](https://user-images.githubusercontent.com/100013964/215492322-7372e15a-8653-43c9-b182-ff589d201667.png)
![Screenshot 2023-01-30 185125](https://user-images.githubusercontent.com/100013964/215492748-b05b7323-f951-40ad-8f95-cb907afc4a46.png)

So here's the PDE that I'm feeding it:
![Screenshot 2023-01-30 191200](https://user-images.githubusercontent.com/100013964/215493677-4eb15d8d-26bd-422a-9734-94b954120d98.png)

And here are the ICs and BCs that I fed it:
![icbc](https://user-images.githubusercontent.com/100013964/215496137-044a82ec-a7ce-4693-b873-20e8fc55c4f4.png)

I'm getting the following result for the density(rho):

![Screenshot 2023-01-30 190255](https://user-images.githubusercontent.com/100013964/215493832-5050d724-7609-4bf1-9de2-4b50a57bc730.png)

While it is able to model the propagation of shock wave roughly, it gets the velocity wrong (1 instead of 0.1). For reference, the correct solution is given as:
![Screenshot 2023-01-30 193136](https://user-images.githubusercontent.com/100013964/215498233-c25ecc40-9107-4b86-8721-58af8f13eab2.png)

When trying a slightly different problem, here's what I'm getting(u=1, in domain(-1,1) with shock initially at 0):

![Screenshot 2023-01-30 191443](https://user-images.githubusercontent.com/100013964/215494175-6d4daa3f-57bf-4b69-af0c-32bfe5e93b09.png)

Here the convergence to solution is better -- however, regardless of the value of u the soln. always seems to converge to the shock wave travelling at 0.1 velocity. Any clues on how to solve this?

Thanks in advance!

**Comment by lululxvi** [comment]
See https://www.sciencedirect.com/science/article/pii/S0045782519306814
